sevsnpmeasure/guest.py

```diff
@@ -101,8 +106,42 @@ def snp_calc_launch_digest(vcpus: int, vcpu_sig: int, ovmf_file: str,
     snp_update_metadata_pages(gctx, ovmf, sev_hashes, vmm_type)
 
     vmsa = VMSA(SevMode.SEV_SNP, ovmf.sev_es_reset_eip(), vcpu_sig, vmm_type)
-    for vmsa_page in vmsa.pages(vcpus):
+    for i, vmsa_page in zip(range(vcpus), vmsa.pages(vcpus)):
+        gctx.update_vmsa_page(vmsa_page)
+        if dump_vmsa is True:
+            with open(f"vmsa{i}.bin", "wb") as f:
+                f.write(vmsa_page)
```

Comment:
a more modern way would be:

```
import pathlib
...
pathlib.Path(f"vmsa{i}.bin").write_bytes(vmsa_page)
```
(not tested)

WDYT?

Reply:
I will try and change it

Reply:
Done